python/ray/data/dataset.py

```diff
@@ -2898,6 +2901,9 @@ def iter_torch_batches(
         batch_size: Optional[int] = 256,
         dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
         device: Optional[str] = None,
+        collate_fn: Optional[
+            Callable[[Union[np.ndarray, Dict[str, np.ndarray]]], "TorchTensorBatchType"]
+        ] = None,
```

Comment:
Should this we allow `collate_fn` to return anything?
```suggestion
        collate_fn: Optional[
            Callable[[Union[np.ndarray, Dict[str, np.ndarray]]], Any]
        ] = None,
```

If we require `torch.Tensor | dict[str, torch.Tensor`, then `collate_fn` won't work for object detection.

Reply:
can you tell me more about the object detection use case?

how would we do the dtype and device conversion if `collate_fn` can return `Any`?

Reply:
ok updated to Any. We are no longer doing dtype and device conversion if collate_fn is passed.